src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -202,46 +209,46 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         def switchPair = switchPairs.all().withAtLeastNNonOverlappingPaths(2).random()
 
         when: "Create 2 not diverse flows going through these switches"
-        def flow1 = flowHelperV2.randomFlow(switchPair)
-        flowHelperV2.addFlow(flow1)
-        def flow1Path = PathHelper.convert(northbound.getFlowPath(flow1.flowId))
-        def flow2 = flowHelperV2.randomFlow(switchPair, false, [flow1])
-        flowHelperV2.addFlow(flow2)
-        def flow2Path = PathHelper.convert(northbound.getFlowPath(flow1.flowId))
+        def flow1 = flowFactory.getRandom(switchPair)
+        def flow1Path = flow1.retrieveAllEntityPaths().getPathNodes()
+        def flow2 = flowFactory.getRandom(switchPair, false, FlowState.UP, flow1.occupiedEndpoints())
+        def flow2Path = flow2.retrieveAllEntityPaths().getPathNodes()
 
         then: "Both flows use the same path"
         flow1Path == flow2Path
 
         when: "Update second flow to become diverse with the first flow (partial update)"
-        flowHelperV2.partialUpdate(flow2.flowId, new FlowPatchV2().tap { diverseFlowId = flow1.flowId })
+        def updateRequest = new FlowPatchV2().tap { diverseFlowId = flow1.flowId }
+        flow2.partialUpdate(updateRequest)
 
         then: "Flows use diverse paths"
-        pathHelper.getInvolvedIsls(flow1.flowId).intersect(pathHelper.getInvolvedIsls(flow2.flowId)).empty
+        def flow1InvolvedIsls = flow1.retrieveAllEntityPaths().flowPath.getInvolvedIsls()
+        def flow2InvolvedIsls = flow2.retrieveAllEntityPaths().flowPath.getInvolvedIsls()
+        flow1InvolvedIsls.intersect(flow2InvolvedIsls).empty
     }
 
     def "Able to do partial update on a single-switch flow"() {
         given: "A single-switch flow"
         def swPair = switchPairs.singleSwitch().random()
-        def flow = flowHelperV2.randomFlow(swPair)
-        flowHelperV2.addFlow(flow)
-        def originalCookies = northbound.getSwitchRules(swPair.src.dpId).flowEntries.findAll {
+        def flow = flowFactory.getRandom(swPair)
+        def originalCookies = switchRulesFactory.get(swPair.src.dpId).getRules().findAll {
             def cookie = new Cookie(it.cookie)
             !cookie.serviceFlag || cookie.type == CookieType.MULTI_TABLE_INGRESS_RULES
         }*.cookie
 
         when: "Request a flow partial update for a 'priority' field"
         def newPriority = 777
         def updateRequest = new FlowPatchV2().tap { it.priority = newPriority }
-        def response = northboundV2.partialUpdate(flow.flowId, updateRequest)
+        def response = flow.partialUpdate(updateRequest)
```

Comment:
the same here about checking response from update request and then flow details

Reply:
Done